api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPractitionerServiceImpl.java

```diff
@@ -92,17 +96,35 @@ public Practitioner create(@Nonnull Practitioner newResource) {
 	@Override
 	@Transactional(readOnly = true)
 	public IBundleProvider searchForPractitioners(PractitionerSearchParams practitionerSearchParams) {
-		IBundleProvider providerBundle = searchQuery.getQueryResults(practitionerSearchParams.toSearchParameterMap(), dao,
-		    translator, searchQueryInclude);
-		SearchParameterMap theParams = new SearchParameterMap();
-		IBundleProvider userBundle = userService.searchForUsers(theParams);
 		
-		if (!providerBundle.isEmpty() && !userBundle.isEmpty()) {
+		IBundleProvider providerBundle = null;
+		IBundleProvider userBundle = null;
+		
+		if (shouldSearchExplicitlyFor(practitionerSearchParams.getTag(), "provider")) {
+			providerBundle = searchQuery.getQueryResults(practitionerSearchParams.toSearchParameterMap(), dao, translator,
+			    searchQueryInclude);
+		}
+		
+		if (shouldSearchExplicitlyFor(practitionerSearchParams.getTag(), "user")) {
+			userBundle = userService.searchForUsers(practitionerSearchParams.toSearchParameterMap());
```

Comment:
Note that formerly the search parameters were not passed into the search for users method, which seemed incorrect, but these parameters may not currently be used in a user context?